ts/parts/Dynamics.ts

```diff
@@ -1040,10 +1040,15 @@ extend(Point.prototype, /** @lends Highcharts.Point.prototype */ {
 
             point.applyOptions(options);
 
-            // Update visuals
-            if (point.y === null && graphic) { // #4146
+            // Update visuals, #4146
```

Comment:
Can we avoid this juggling of conditions? Under what circumstances do we create a dummy graphic? Is it always added when the a11y module is loaded? In that case, maybe we can have a series property `series.useDummyGraphics`, and simplify the condition like so:

```js
if (point.y === null && graphic && !series.useDummyGraphics) {
    point.graphic = graphic.destroy();
}
```

Reply:
The dummy graphics are added for null points only - to ensure points always have a graphic if they should be announced to screen readers. So this condition needs to destroy graphics for:
- Points that are no longer null that have a dummy graphic
- Points that are null now and no longer should have a graphic, unless it has a dummy graphic

So it could be
``` javascript
// Option #1
if (
  graphic && (
    point.y !== null && point.hasDummyGraphic ||
    point.y === null && !point.hasDummyGraphic
  )
)
```
or slightly more explicit for future readers, the `shouldDestroyGraphic` definition could be replaced with
```javascript
// Option #2
const shouldDestroyGraphic = point.y !== null && hasDummyGraphic ||
    point.y === null && !hasDummyGraphic
```
which accomplishes the same as in the PR, but is a bit longer. Minification will reduce it down regardless, so it depends on what we find readable. I have no strong opinions either way, maybe leaning slightly towards Option 1 since it is pretty clear already that we are destroying the graphic.

Reply:
Sorry to interrupt, but did you consider not using `point.graphic`? Something like `point.a11nGraphic` instead? Something you can freely add/remove in `describePointsInSeries`.

Then simple change change in `point.destroy` from 
```
            // Remove all events and elements
            if (point.graphic || point.dataLabel || point.dataLabels) {
                removeEvent(point);
                point.destroyElements();
            }
```
To:
```
            // Remove all events and elements
            if (point.shouldDestroyGraphic()) {
                removeEvent(point);
                point.destroyElements();
            }
```
Could help solving the issue? It's not the first time we wish to add a graphic-like element to a point which will be automatically removed (even when point is null) on point remove.

Reply:
Yes, it was considered, and might still be a viable option. However because a point should never have a dummy graphic and a real graphic at the same time, it will lead to more changes in the core - managing the lifecycle of the dummy graphic.

The dummy graphic will have to be managed in `point.update` as well as in `point.destroy`, and it will cause a domino effect in the a11y module having to look for `point.dummyGraphic` everywhere `point.graphic` does not exist. If `point.graphic` is overwritten at any point in the code (now or in the future), the `point.dummyGraphic` should also be removed, which could be a source of bugs.

Reply:
Okay, I see it is not as simple as I thought.